code/procs/helpers.dm

```diff
@@ -1879,6 +1879,9 @@ proc/countJob(rank)
 	if (!allow_dead_antags && (!G.mind || G.mind && (G.mind.dnr || !isnull(G.mind.special_role) || G.mind.former_antagonist_roles.len))) // Dead antagonists have had their chance.
 		return 0
 
+	if (G.mind && G.mind.dnr)
+		return 0
```

Comment:
Should probably just move this up to the top since there's no reason to bother with any of these checks if they were set to DNR anyways. Also probably shouldn't just proceed to spawn people who don't have minds just because they are also a traitor. Just wrap it in with the initial checks

```dm
	if (!G?.mind || G.mind.dnr)
		return 0
```

don't need the explicit `G` check because thats built in now. Likewise only mobs have minds so checking `ismob(G)` is pretty pointless since bad input would have a null mind anyway.